CenterForOpenScience/osf.io
# [QA Prep Required] Feature/Fix EmbargoTerminationApproval Status on Auto-Approve [ENG-463]

## Purpose

If your registration is embargoed (not going to become public for a set amount of time), the user has the option to end that embargo early.  If that process is initiated, but all the required users haven't approved this early embargo termination, it will be automatically approved after 48 hours by a script that's run nightly.

This script is marking the Registration as `public`, and marking the Embargo as `completed`, but failing to mark the EmbargoTerminationApproval as `approved`.  This puts the API in a weird state. When accessing a `public_registration`, its `pending_embargo_termination_approval` field is True (it should be False).  This puts the front end in a weird state by showing a banner that shouldn't be visible. 

## Changes

- Fix the `scripts/approve_embargo_terminations.py` script that runs nightly to correctly mark the state of affected EmbargoTerminationApproval as `approved`.
- Add a temporary script to fix bad data (between 100-200 registrations)
- Add a test to ensure this field is being set properly

## QA Notes
- Make some registrations that are embargoed.  Approve the embargo.  Attempt to set the registration public. Instead of approving the Embargo Termination Request, wait for the cron job to automatically approve it.  Verify that when visiting the registration in the API, you see `is_public=True`, but `pending_embargo_termination_approval` is also True. You should also see a weird banner in the UI.  

After the script has run to correct the migrations, verify that these bad registrations have `pending_embargo_termination_approval` set to False in the API, and that the banner is gone from the registration.

Verify that repeating this process for new embargoes (after this work is merged), the API state is correct.
## Documentation



## Side Effects



## Ticket

https://openscience.atlassian.net/browse/ENG-463
